### PR TITLE
Minimize type parameters

### PR DESCRIPTION
**Pull request by LoiNguyenCS**
Professor,

This is the PR to remove unused type bounds of type parameters in the declaration of a class. After this PR and #101 are merged, Specimin will be able to minimize [AnalysisResult.java](https://github.com/mernst/checker-framework/blob/master/dataflow/src/main/java/org/checkerframework/dataflow/analysis/AnalysisResult.java).

### CODE REVIEW

File: src/test/java/org/checkerframework/specimin/UnusedTypeParameterBoundTest.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package org.checkerframework.specimin;
+
+import java.io.IOException;
+import org.junit.Test;
+
+/**
+ * This test checks if Specimin can remove unused bounds of type parameters in the declaration of a
+ * class.
```
Comment (kelloggm) [inline src/test/java/org/checkerframework/specimin/UnusedTypeParameterBoundTest.java:8]: Can you please add a test for generic type bounds that should not be removed, as well. We don't currently have one.

Comment (kelloggm) [reply src/test/java/org/checkerframework/specimin/UnusedTypeParameterBoundTest.java:8]: (to check that, I searched our `src/test/resources/` directory for "extends", and then observed that none of the results are type variables)

**Review by kelloggm — CHANGES_REQUESTED** [review]

**Comment by LoiNguyenCS** [comment]
Professor,

I have updated the test case to remove one generic type bound and preserve the other type bound.

**Review by kelloggm — APPROVED** [review]
Thanks for the new test case. LGTM!

**Pull request merged**